tests/core/test_block_manager.py

```diff
@@ -90,6 +90,38 @@ def test_allocate():
     assert block_manager.can_allocate(seq_group) != AllocStatus.OK
 
 
+def test_allocate_encoder_decoder():
+    block_size = 4
+    num_cpu_blocks = 4
+    num_gpu_blocks = 4
+    block_req_per_seq_group = 2
+    block_manager = BlockSpaceManagerV1(block_size,
+                                        num_cpu_blocks,
+                                        num_gpu_blocks,
+                                        watermark=0)
+
+    # Allocate same sequence group to all available gpu blocks.
+    for i in range(num_gpu_blocks // block_req_per_seq_group):
+        _, _, seq_group = create_dummy_prompt_encoder_decoder(
+            str(i), block_size, block_size)
+        assert block_manager.can_allocate(seq_group)
```

Comment:
Was this intended to assert that the allocation is possible like this?
```
assert block_manager.can_allocate(seq_group) == AllocStatus.OK
```

IIUC asserting on the result of `block_manager.can_allocate` will never fail because it always returns a value from the `AllocStatus` enum. This doesn't raise for me:
```
from vllm.core.interfaces import AllocStatus

assert AllocStatus.OK
assert AllocStatus.LATER
assert AllocStatus.NEVER
```

Reply:
This bug was actually carried over from the pre-existing decoder-only test_allocate() unit test which I copied; I will shortly push a commit which fixes this bug in both the decoder-only and encoder/decoder tests.

Great catch!